File: app/src/main/java/com/ebsoft/watchlist/data/DataManagerImpl.java
package com.ebsoft.watchlist.data;

import com.ebsoft.watchlist.data.local.db.DBManager;
import com.ebsoft.watchlist.network.APIManager;

import javax.inject.Inject;
import javax.inject.Singleton;

/**
 * Created by endre on 08/09/18.
 */

@Singleton
public class DataManagerImpl implements DataManager {

    private final APIManager mApiManager;

    private final DBManager mDbManager;

    @Inject
    public DataManagerImpl(APIManager apiManger, DBManager dbManager) {
        mApiManager = apiManger;
        mDbManager = dbManager;
    }

    @Override
    public APIManager getApiManager() {
        return mApiManager;
    }

    @Override
    public DBManager getDbManager() {
        return mDbManager;
    }
}

File: app/src/main/java/com/ebsoft/watchlist/data/DataManagerImpl.kt
package com.ebsoft.watchlist.data

import com.ebsoft.watchlist.data.control.db.DBManager
import com.ebsoft.watchlist.data.control.network.APIManager

import javax.inject.Inject
import javax.inject.Singleton

/**
 * Created by endre on 08/09/18.
 */

@Singleton
class DataManagerImpl @Inject
constructor(override val apiManager: APIManager, override val dbManager: DBManager) : DataManager
